### ISSUE TITLE
Feature: Scan for service classes inside servlets annotated with @WebServlet (ServletAPI v3+)

### ISSUE THREAD

**Issue by medvedev** [GoogleCloudPlatform/endpoints-framework-gradle-plugin#59]
Hi,

Servlet API v3 allows annotations-only config with "WebServlet" and "WebInitParam" annotations on servlet class.

Currently endpoints-framework-plugin does not support scan of classpath to extract metadata from "WebServlet" annotations and use it inside endpointsOpenApiDocs task.

I've added some code that introduces described functionality. [Here it is](https://github.com/medvedev/endpoints-framework-gradle-plugin/commit/1eadd7ad14959e6a2731943d7c5fcfca47349109)
Please take a look and tell if it's ok to file a pull request or there are major changes needed.

**Comment by patflynn** [comment]
Thanks @medvedev for the detailed issue.  We're considering this and @loosebazooka should get back to you soon.

**Comment by medvedev** [comment]
@patflynn, A little more background.
I'm relatively new to appengine and am not sure if I've picked right place to add annotations scan.
The initial Issue I've faced was:
When I create new appengine SE app with java8 and servlet 3.0 annotations (without web.xml) and run any endpoints generate task from this plugin, I get "usage" output (and "BUILD SUCCESSFUL" in the end!). That's confusing IMO.
So, I started digging deeper and found out that in my case it's necessary to list all API classes twice (in @WebService annotation and build.gradle), then got some time and inspiration for attempt to fix it.

The separate pull request that doesn't contain any non-mandatory refactoring: #61
